### PR TITLE
add RProject.Rtools 4.0.1

### PR DESCRIPTION
**Pull request by dpprdan**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/16352)

### CODE REVIEW

File: manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.installer.yaml
Comment (ItzLevvie) [inline manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.installer.yaml:3]:
```suggestion
PackageVersion: 4.0
```
image

File: manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.locale.en-US.yaml
Comment (ItzLevvie) [inline manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.locale.en-US.yaml:3]:
```suggestion
PackageVersion: 4.0
```
image

File: manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.locale.en-US.yaml
Comment (ItzLevvie) [inline manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.locale.en-US.yaml:10]:
```suggestion
PackageName: Rtools
```

File: manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.yaml
Comment (ItzLevvie) [inline manifests/r/RProject/Rtools/4.0.1/RProject.Rtools.yaml:3]:
```suggestion
PackageVersion: 4.0
```
image

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
​

**Comment by dpprdan** [comment]
@ItzLevvie I am not sure about the `PackageVersion: 4.0` as opposed to `PackageVersion: 4.0.1`, since this is a [new version of Rtools4](https://twitter.com/opencpu/status/1395692067884503044) and 4.0.0 [already exists](https://github.com/microsoft/winget-pkgs/blob/67ecd21017fc62b8e6ac6f836bb63a6b7ab96e61/manifests/r/RProject/Rtools/4.0.0/RProject.Rtools.yaml#L4) in winget-pkgs. So `4.0` seems confusing IMHO. What do you think?

**Comment by dpprdan** [comment]
@jeroen Would it be possible to use semantic versioning for Rtools everywhere? On https://cran.r-project.org/bin/windows/Rtools/ the latest version's file name is `rtools40v2-x86_64.exe` and in Windows > Apps & Features it shows up as `Rtools 4.0 (4.0.1.0)`. This is a slightly confusing, see above.

screenshot with Rtools 4.0 (4.0.1.0)

IMHO it would be better to name it `rtools-4.0.1-x86_64.exe` and `Rtools 4.0.1`, respectively (cf `R-4.1.0-win.exe` for [base R](https://cran.r-project.org/bin/windows/base/)).

**Comment by ItzLevvie** [comment]
The issue with keeping it `4.0.1` is that when someone does `winget upgrade` on their machine it'll always keep updating it even if you're on the latest version because 4.0.1 is newer than what's shown in Control Panel (and you'll end up in an endless loop of updating.)

**Comment by jeroen** [comment]
I don't think calling it 4.0.1 is a good idea because that may be confused that this is related to the R version 4.0.1. This is not the case. Rtools 4.0 can be used with any R >= 4.0.

The `.1` is the built number postfix. Perhaps I should use a date there, just like msys2. So that it becomes `4.0.20210604`

**Review by ItzLevvie — CHANGES_REQUESTED** [review]
​

**Comment by jeroen** [comment]
@ItzLevvie thanks! An Rtools 4.0 package is indeed the proper solution

**Comment by dpprdan** [comment]
Updating the manifest seems to be the right solution in this particular case since the version tag is not properly set in the installer. 

In general, though, I still think it would be better to have a manifest for every (proper) version. At least this is how I read @denelon's [advice](https://github.com/microsoft/winget-pkgs/issues/9476#issuecomment-811158348) to "create a new manifest (assuming the URL is unique to a specific version)" for new versions of packages and "to retain manifests for every valid version that may still be installed". 

Older versions of RTools are still available back to RTools26 [at the source](https://cran.r-project.org/bin/windows/Rtools/history.html), Chocolatey for example also [lists multiple versions](https://community.chocolatey.org/packages/rtools/3.5.0.4#releasenotes). Why should this be any different with winget? The default `winget install rtools` command (i.e. without the `--version` tag) or `winget upgrade` ensure that users get the recommended latest version. But if they need to get an old version for e.g. debugging they still can.

@ItzLevvie While I agree with the solution, I have to say that I am puzzled by your style. Maybe it's just me, but it feels quite insincere to have the moderator open another PR (#16740 by the way) without notice instead of waiting for the OP to react to and fixing the issue. Even more so when it's done with a second account with the stated goal "to merge my own PRs to make my changes available to production as soon as possible." Is this a race somehow?

grafik

This feels like volunteer contributions are not valued very much here. I am not sure whether this is the sentiment you want to convey.

**Comment by ItzLevvie** [comment]
I'm sorry if that's what you think - anyone is free to contribute to the catalog and that's why this repository was open-source to begin with. Without anyone's help, it wouldn't be a "package manager" as it wouldn't have that many applications.

The second account wasn't made to "race" with anyone. It was created merely to create "own" PRs and get it merged faster into master without waiting for another moderator's approval as I put out hundreds of PRs to fix, update or remove manifests and it may take hours or days for it to be merged, not to mention that it also has to be tested to see if it shows up currently in `winget list`, etc.

I'm in the process of installing 1900+ applications on a virtual machine and fixing some of the versions, names, and publishers, so even if the account and this PR wasn't created, the problem with this manifest would've been noticed by me and fixed.

It was only several weeks ago that I realized how important it is to update values to match what's shown in Control Panel's ARP. Unfortunately, I don't use Chocolatey, so I don't know how their system works in terms of what they do to see if a particular application is installed, so it's not really my fault if application developers aren't showing a proper version or updating it, etc.

**Comment by dpprdan** [comment]
> I'm sorry if you feel that way, and not notifying you about the PR I had created was a mistake on my end and I deeply apologize for that.

Thank you, that's all I needed to hear! 😃 

As for the second account: I understand the reasons for creating it. This workaround just looks weird from the outside and it sounds to me like processes are not optimal yet. But I am sure you and the winget team will find a good solution here.

Thanks for your work!

**Comment by denelon** [comment]
@dpprdan we are working to improve all of the end to end scenarios. Moderation is a very recent necessity to help ensure a high quality catalog of packages in the community repository.  At present, there are only three community moderators fielding hundreds of submissions from the community, and trying to keep up with "Help-Wanted" asks. We have been balancing direct community support against fixing bugs, and building new features. These are exciting times.

We sincerely appreciate contributions!
* Feedback
* Feature Requests
* Package Submissions
* Manifest improvements
* Bug Reports
* Discussions
* PRs on documentation
* PRs on code

The last couple of weeks have been a bit chaotic with the number of new users and newly identified edge cases. Many of them were known, but many of them and several new ones have finally become reproducible. 

Feedback is critical to our success. We believe in transparency and we rely on support from the community of customers, publishers, and ISVs to help us build the best product possible.

**Pull request closed without merging**
